core/src/test/java/org/apache/iceberg/hadoop/HadoopTableTestBase.java

```diff
@@ -170,10 +172,15 @@ void rewriteMetadataAsGzipWithOldExtension() throws IOException {
   }
 
   protected HadoopCatalog hadoopCatalog() throws IOException {
+    return hadoopCatalog(Collections.emptyMap());
+  }
+
+  protected HadoopCatalog hadoopCatalog(Map<String, String> catalogProperties) throws IOException {
     HadoopCatalog hadoopCatalog = new HadoopCatalog();
     hadoopCatalog.setConf(new Configuration());
     hadoopCatalog.initialize("hadoop",
-            ImmutableMap.of(CatalogProperties.WAREHOUSE_LOCATION, temp.newFolder().getAbsolutePath()));
+        ImmutableMap.<String, String>builder().putAll(catalogProperties).put(CatalogProperties.WAREHOUSE_LOCATION,
+            temp.newFolder().getAbsolutePath()).build());
```

Comment:
Style: In Iceberg, try to wrap lines at the highest semantic level possible, rather than trying to use the whole line. In this case, it is awkward that `temp.newFolder()...` is on its own line because it's inside a method call.

I'd change it to this:

```java
    hadoopCatalog.initialize("hadoop",
        ImmutableMap.<String, String>builder()
            .putAll(catalogProperties)
            .put(CatalogProperties.WAREHOUSE_LOCATION, temp.newFolder().getAbsolutePath())
            .build());
```